src/main/java/de/adesso/blogpostchecker/FileAnalyzer.java

```diff
@@ -0,0 +1,180 @@
+package de.adesso.blogpostchecker;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.ListBranchCommand;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.diff.DiffEntry;
+import org.eclipse.jgit.diff.DiffFormatter;
+import org.eclipse.jgit.lib.*;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.treewalk.CanonicalTreeParser;
+import org.eclipse.jgit.treewalk.TreeWalk;
+import org.gitective.core.BlobUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@Service
+public class FileAnalyzer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FileAnalyzer.class);
+
+    private ConfigService configService;
+
+    private boolean analyzed;
+    private PostHeader header;
+    private String authors;
+
+    @Autowired
+    public FileAnalyzer(ConfigService configService) {
+        this.configService = configService;
+    }
+
+    public PostHeader getHeader() {
+        if (!analyzed) {
+            analyzeBranch();
+        }
+        return header;
+    }
+
+    public String getAuthors() {
+        if (!analyzed) {
+            analyzeBranch();
+        }
+        return authors;
+    }
+
+    private void analyzeBranch() {
+        try {
+            Git localGit = LocalRepoCreater.getLocalGit();
+            Repository localRepo = localGit.getRepository();
```

Comment:
```suggestion
            Repository localRepository = localGit.getRepository();
```
Clean Code sagt Kürzungen sind zu vermeiden. Ausschreibungen sind zu bevorzugen :)
+ Einheitlichkeit
+ Lesbarkeit
+ Aussprechbarkeit